test/e2e_node/restart_test.go

```diff
@@ -265,5 +269,103 @@ var _ = SIGDescribe("Restart [Serial] [Slow] [Disruptive]", func() {
 				}
 			}
 		})
+		// Regression test for https://issues.k8s.io/116925
+		ginkgo.It("should delete pods which are marked as terminal and have a deletion timestamp set after restart", func(ctx context.Context) {
```

Comment:
I think we should mark the test as Disruptive and Serial because it restarts Kubelet. In the past I observed that kubelet restarts make other tests fail occasionally.

Reply:
Yes, this is already done, please see https://github.com/kubernetes/kubernetes/pull/117019#discussion_r1162479722